packages/lesswrong/components/ea-forum/giving-portal/hooks.ts

```diff
@@ -17,19 +21,34 @@ export type ElectionAmountRaisedQueryResult = {
 }
 
 export const useElectionCandidates = (
-  sortBy: ElectionCandidatesSort = "mostPreVoted",
+  sortBy: ElectionCandidatesSort | "random" = "mostPreVoted",
   options?: Partial<UseMultiOptions<"ElectionCandidateBasicInfo", "ElectionCandidates">>,
 ) => {
-  return useMulti({
+  const currentUser = useCurrentUser();
+  const [cookies] = useCookiesWithConsent([CLIENT_ID_COOKIE]);
+  const clientId = cookies[CLIENT_ID_COOKIE];
+
+  const {results, ...retVal} = useMulti({
     collectionName: "ElectionCandidates",
     fragmentName: "ElectionCandidateBasicInfo",
     terms: {
       electionName: eaGivingSeason23ElectionName,
-      sortBy,
+      sortBy: sortBy === "random" ? "name" : sortBy,
     },
     limit: 30,
     ...options,
   });
+
+  let resultsCopy = results ? [...results] : undefined;
+  if (sortBy === "random") {
+    const rng = seedrandom(currentUser?.abTestKey ?? clientId);
+    resultsCopy?.sort(() => rng() - 0.5);
+  }
```

Comment:
Another thing that wouldn't be ideal if we were maintaining this for a long time, but given that we know the list is only 25 it's fine to fetch them all and sort them on the frontend.

This also adds random as an option in the giving portal, I can make it the default there if desired (cc @agnesstenlund )

Reply:
Random as default seems good for the portal too, thanks!